ui/StatusQ/src/oneoffilter.cpp

```diff
@@ -0,0 +1,114 @@
+#include "StatusQ/oneoffilter.h"
+
+#include <QTimer>
+
+/*!
+    \qmltype OneOfFilter
+    \inherits RoleFilter
+    \inqmlmodule SortFilterProxyModel
+    \ingroup Filters
+    \brief Filters rows included in the \p array
+
+    A OneOfFilter is a \l RoleFilter that accepts rows if their data is contained in the \p array property.
+    The values in \p array are converted to strings.
+    If the \p separator is supplied, the \p array is treated as one string, separated by it.
+
+    \code
+    readonly property string filterArray1: "1:10:42" // needs the ":" separator param
+    readonly property var filterArray2: ["1", "10", "42"]
+
+    SortFilterProxyModel {
+      sourceModel: chainsModel
+      filters: OneOfFilter {
+        array: filterArray1 // or filterArray2
+        separator: ":" // <-- optional
+      }
+    }
+    \encode
+*/
+
+OneOfFilter::OneOfFilter(QObject *parent)
+    : RoleFilter(parent)
+{
+}
+
+bool OneOfFilter::filterRow(const QModelIndex& sourceIndex, const qqsfpm::QQmlSortFilterProxyModel& proxyModel) const
+{
+    const auto strValue = sourceData(sourceIndex, proxyModel).toString();
+    return m_actualArray.contains(strValue);
+}
+
+void OneOfFilter::proxyModelCompleted(const qqsfpm::QQmlSortFilterProxyModel &proxyModel)
+{
+    if (roleName().isEmpty()) {
+        qWarning() << Q_FUNC_INFO << "Required property 'roleName' is not set";
+        return;
+    }
+    updateActualArray();
+
+    connect(this, &OneOfFilter::paramsChanged, this, &OneOfFilter::updateActualArray);
+    connect(this, &OneOfFilter::actualArrayChanged, this, &OneOfFilter::invalidate);
+}
+
+QVariant OneOfFilter::array() const
+{
+    return m_array;
+}
+
+void OneOfFilter::setArray(const QVariant &newArray)
+{
+    if (m_array == newArray)
+        return;
+    m_array = newArray;
+    emit paramsChanged();
+}
+
+QString OneOfFilter::separator() const
+{
+    return m_separator;
+}
+
+void OneOfFilter::setSeparator(const QString &newSeparator)
+{
+    if (m_separator == newSeparator)
+        return;
+    m_separator = newSeparator;
+    emit paramsChanged();
+}
+
+QStringList OneOfFilter::actualArray() const
+{
+    return m_actualArray;
+}
+
+void OneOfFilter::updateActualArray()
+{
+    if (m_array.isNull() || !m_array.isValid()) {
+        qWarning() << Q_FUNC_INFO << "Supplied 'array' is empty or invalid!" << m_array;
```

Comment:
The message mentions empty array, but the condition checks for `isNull` and `isValid`. Empty but well-formed array shouldn't trigger that and is valid value, right?